Update CI to use Loki 2.4.2

vyzigold opened the pull request.

vyzigold committed:
Update CI to use Loki 2.4.2

paramite commented:
We need to update test image from centos:8 to quay.io/centos/centos:stream8 or even :stream9 probably

paramite commented:
Can you make that change as part of this PR please?

vyzigold committed:
Replace loki config

source: https://github.com/grafana/loki/blob/main/cmd/loki/loki-docker-config.yaml

vyzigold committed:
Replace centos image

vyzigold committed:
Wait for loki to become ready

vyzigold committed:
Fix Loki test queries

vyzigold commented:
I updated centos to stream8. I tried updating to steam9, but I wasn't able to install qpid-proton-c-devel from anywhere, not even epel.
